jerry-core/vm/vm.c

```diff
@@ -2010,10 +2010,49 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         {
           if (ecma_are_values_integer_numbers (left_value, right_value))
           {
-            ecma_integer_value_t left_integer = (ecma_integer_value_t) left_value;
-            ecma_integer_value_t right_integer = (ecma_integer_value_t) right_value;
+            bool is_less = (ecma_integer_value_t) left_value < (ecma_integer_value_t) right_value;
+
+            if (*byte_code_p <= CBC_BRANCH_IF_TRUE_BACKWARD_3 && *byte_code_p >= CBC_BRANCH_IF_TRUE_BACKWARD)
+            {
+              byte_code_start_p = byte_code_p++;
+              branch_offset = 0;
+
+              switch (CBC_BRANCH_OFFSET_LENGTH (*byte_code_start_p))
+              {
+                case 1:
+                {
+                  branch_offset = *(byte_code_p++);
+                  break;
+                }
+                case 3:
+                {
+                  branch_offset = *(byte_code_p++);
+                  /* FALLTHRU */
+                }
+                default:
+                {
+                  JERRY_ASSERT (CBC_BRANCH_OFFSET_LENGTH (*byte_code_start_p) == 2
+                                || CBC_BRANCH_OFFSET_LENGTH (*byte_code_start_p) == 3);
+
+                  branch_offset <<= 8;
```

Comment:
If this approach is kept, then this statement can be hoisted to `case 3` above. (If `case 2`, there is no point in shifting 0 left by 8 bits.)